### DISCUSSION TITLE
Why focal_length in viewport_height?

### DISCUSSION THREAD

**Discussion by xrailab** [RayTracing/raytracing.github.io#1299] [Q&A]
Anyone explain to me why, in the section Positionable Camera, we have in the code

auto viewport_height = 2 * h * focal_length;

In particular, why do we multiply focal_lenght? I am sure it is important to have a viewport dimension that depends on focal_lenght , but I cannot realize the modelling rationale.

**Comment by hollasch** [comment]
You must multiply by the focal length in order to preserve the desired viewing angle. $tan(\theta) = y/x$, so $y = x \cdot tan(\theta)$. Here, x ⇔ `focal_length`.
